server/models/postgis/project.py

```diff
@@ -382,26 +382,53 @@ def get(project_id: int):
 
     def update(self, project_dto: ProjectDTO):
         """ Updates project from DTO """
-        self.status = ProjectStatus[project_dto.project_status].value
-        self.priority = ProjectPriority[project_dto.project_priority].value
-        self.default_locale = project_dto.default_locale
-        self.mapping_permission = MappingPermission[
-            project_dto.mapping_permission.upper()
-        ].value
-        self.validation_permission = ValidationPermission[
-            project_dto.validation_permission.upper()
-        ].value
-        self.enforce_random_task_selection = project_dto.enforce_random_task_selection
-        self.private = project_dto.private
-        self.mapper_level = MappingLevel[project_dto.mapper_level.upper()].value
-        self.entities_to_map = project_dto.entities_to_map
-        self.changeset_comment = project_dto.changeset_comment
-        self.due_date = project_dto.due_date
-        self.imagery = project_dto.imagery
-        self.josm_preset = project_dto.josm_preset
-        self.id_presets = project_dto.id_presets
+        project_dto_data = self._get_project_and_base_dto()[1]
+        updated_fields = [
+            key for key, value in project_dto.items() if value is not None and value
+        ]
+        [
+            setattr(project_dto, key, value)
+            for key, value in project_dto_data.items()
+            if key not in updated_fields
+        ]
+
+        if project_dto.project_status:
+            self.status = ProjectStatus[project_dto.project_status].value
+        if project_dto.project_priority:
+            self.priority = ProjectPriority[project_dto.project_priority].value
+        if project_dto.default_locale:
+            self.default_locale = project_dto.default_locale
+        if project_dto.mapping_permission:
+            self.mapping_permission = MappingPermission[
+                project_dto.mapping_permission.upper()
+            ].value
+        if project_dto.validation_permission:
+            self.validation_permission = ValidationPermission[
+                project_dto.validation_permission.upper()
+            ].value
+        if project_dto.enforce_random_task_selection:
+            self.enforce_random_task_selection = (
+                project_dto.enforce_random_task_selection
+            )
+        if project_dto.private:
```

Comment:
that line is avoiding to update the project to `private = False`

Reply:
Thanks for the review!

I will try to fix all the problems.

As soon as I find a solution I comment here

Reply:
I changed the logic to get the fields to update, removed the if statement for the `changeset_comment` and `private` fields and left a comment explaining that these fields don't have a comparison because it would affect when changed for `False` or an empty string.

Get updated fields line
```py
updated_fields = [
        key
        for key, value in project_dto.items()
        if value is not None and project_dto_data[key] != value
]
```

`changeset_comment` and `private` fields

```py
# Doesn't check if exists because the boolean False when modified affects comparisons
self.private = project_dto.private
if project_dto.mapper_level:
    self.mapper_level = MappingLevel[project_dto.mapper_level.upper()].value
if project_dto.entities_to_map:
    self.entities_to_map = project_dto.entities_to_map
# Doesn't check if exists because empty strings when modified affects comparisons
self.changeset_comment = project_dto.changeset_comment
```